qiskit/algorithms/eigen_solvers/vqd.py

```diff
@@ -0,0 +1,824 @@
+logger = logging.getLogger(__name__)
+
+# disable check for ansatzes, optimizer setter because of pylint bug
+# pylint: disable=no-member
+
+
+class VQD(VariationalAlgorithm, Eigensolver):
+    r"""The Variational Quantum Deflation algorithm.
+
+    `VQD <https://arxiv.org/abs/1805.08138>`__ is a quantum algorithm that uses a
+    variational technique to find
+    the k eigenvalues of the Hamiltonian :math:`H` of a given system.
+
+    An instance of VQD requires defining three algorithmic sub-components:
+    integer k denoting the number of eigenstates to calculate, a trial
+    state (a.k.a. ansatz)which is a :class:`QuantumCircuit`,
+    and one of the classical :mod:`~qiskit.algorithms.optimizers`.
+    The ansatz is varied, via its set of parameters, by the optimizer,
+    such that it works towards a state, as determined by the parameters
+    applied to the ansatz, that will result in the minimum expectation values
+    being measured of the input operator (Hamiltonian). The algorithm does
+    this by iteratively refining each excited state to be orthogonal to all
+    the previous excited states.
+
+    An optional array of parameter values, via the *initial_point*, may be provided as the
+    starting point for the search of the minimum eigenvalue. This feature is particularly useful
+    such as when there are reasons to believe that the solution point is close to a particular
+    point.
+
+    The length of the *initial_point* list value must match the number of the parameters
+    expected by the ansatz being used. If the *initial_point* is left at the default
+    of ``None``, then VQE will look to the ansatz for a preferred value, based on its
+    given initial state. If the ansatz returns ``None``,
+    then a random point will be generated within the parameter bounds set, as per above.
+    If the ansatz provides ``None`` as the lower bound, then VQE
+    will default it to :math:`-2\pi`; similarly, if the ansatz returns ``None``
+    as the upper bound, the default value will be :math:`2\pi`.
+
+    """
+
+    def __init__(
+        self,
+        ansatz: Optional[QuantumCircuit] = None,
+        k: int = 2,
+        betas: Optional[List[float]] = None,
+        optimizer: Optional[Optimizer] = None,
+        initial_point: Optional[np.ndarray] = None,
+        gradient: Optional[Union[GradientBase, Callable]] = None,
+        expectation: Optional[ExpectationBase] = None,
+        include_custom: bool = False,
+        max_evals_grouped: int = 1,
+        callback: Optional[Callable[[int, np.ndarray, float, float], None]] = None,
+        quantum_instance: Optional[Union[QuantumInstance, BaseBackend, Backend]] = None,
+    ) -> None:
+        """
+
+        Args:
+            ansatz: A parameterized circuit used as Ansatz for the wave function.
+
+            optimizer: A classical optimizer.
+            k: the number of eigenvalues to return.
+            beta: beta parameter in the VQD paper. Should have same length as
```

Comment:
I've patched up the docstring. Any suggestions on what might be a better name? The paper leaves it at  'a  hyperparamter, beta' @woodsp-ibm @paulineollitrault @Cryoris

Reply:
Some thoughts, Maybe a very brief description with the math for VQD could be added to the main docstring, so beta is shown there, and which can refer to the paper. Here perhaps it can refer back to that or very brief summary about what it does. It should also be noted that when left at the default of None values are automatically computed and maybe outline what that is here. As to a better name, it correlates to the paper and with a better description here maybe that will be ok. Or maybe filling in a better description here may help in identifying what might be a more descriptive name